src/components/Sidebar/BranchTransition.tsx

```diff
@@ -1,22 +1,75 @@
-import React, { FC, memo } from 'react';
+import React, { FC, memo, useState, useCallback, ChangeEvent } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faPlus, faTools } from '@fortawesome/free-solid-svg-icons';
-
+import DropDown from 'components/elements/DropDown';
+import { TextField, FormControl } from '@material-ui/core';
+import { setTransitionCondition, setTransitionConditionDescription } from 'utils/builder';
 import { SidebarHeader, SidebarButton } from '.';
 import { Pathway, Transition } from 'pathways-model';
+import { useCriteriaContext } from 'components/CriteriaProvider';
 import useStyles from './styles';
 
 interface BranchTransitionProps {
   pathway: Pathway;
+  currentNodeKey: string;
   transition: Transition;
   updatePathway: (pathway: Pathway) => void;
 }
 
-const BranchTransition: FC<BranchTransitionProps> = ({ pathway, transition, updatePathway }) => {
+const BranchTransition: FC<BranchTransitionProps> = ({
+  pathway,
+  currentNodeKey,
+  transition,
+  updatePathway
+}) => {
+  const { criteria } = useCriteriaContext();
+  const criteriaOptions = criteria.map(c => ({ value: c.label, label: c.label }));
   const styles = useStyles();
   const transitionKey = transition?.transition || '';
   const transitionNode = pathway.states[transitionKey];
+  const [useCriteriaSelected, setUseCriteriaSelected] = useState<boolean>(false);
+
+  const handleUseCriteria = useCallback((): void => {
+    setUseCriteriaSelected(!useCriteriaSelected);
+  }, [useCriteriaSelected]);
+
+  const selectCriteriaSource = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      if (!currentNodeKey || !transition.id) return;
+
+      const criteriaSource = event?.target.value || '';
+      let elm = undefined;
+      criteria.forEach(c => {
+        if (c.label === criteriaSource) {
+          elm = c.elm;
+        }
+      });
+      if (!elm) return;
+      updatePathway(
+        setTransitionCondition(
+          pathway,
+          currentNodeKey,
+          transition.id,
+          transition.condition?.description || '',
+          elm,
+          criteriaSource
+        )
+      );
+    },
+    [currentNodeKey, transition.id, updatePathway, pathway, transition.condition, criteria]
+  );
 
+  const setCriteriaDisplay = useCallback(
```

Comment:
Right now each criteria in the provider has only one statement. This would be a backlog item but what do you think about having this set a property in the criteria itself. Then if it already exists it would pre-populate the text field.

Reply:
That could be interesting, We would have to decide if something was already in the description but the criteria label was changing what happens.

Reply:
Yea that's a good point... I'll throw it to Jim & Noranda and see what the use case is. I assumed it would always be the same but I guess that may be too strict. There's nothing wrong with how you're doing it here but if the label is always the same then it would potentially save some duplicate typing